### PR TITLE
Handling requeue and template

### PR DESCRIPTION
**Pull request by mnkg561**
Signed-off-by: nmogulla <mnkg561@gmail.com>

close https://github.com/keikoproj/alert-manager/issues/4

**Could you share the solution in high level?**
1. Write a reusable Update status function to requeue the request in case of an error
2. Add a method to get K8s secret
3. 

**Could you share the test results?**

Unit test cases are working fine

### CODE REVIEW

File: controllers/common/common.go [hunk 1]
```diff
@@ -60,24 +64,24 @@ func (r *Client) UpdateMeta(ctx context.Context, object client.Object) {
...
-//
-//	log.Info("Requeue time", "time", requeueTime[0])
-//	return ctrl.Result{RequeueAfter: time.Duration(requeueTime[0]) * time.Millisecond}, nil
-//}
+func (r *Client) UpdateStatus(ctx context.Context, obj client.Object, state alertmanagerv1alpha1.State, requeueTime ...float64) (ctrl.Result, error) {
+	log := log.Logger(ctx, "controllers.common", "common", "UpdateStatus")
+
+	if err := r.Status().Update(ctx, obj); err != nil {
+		log.Error(err, "Unable to update status", "status", state)
+		r.Recorder.Event(obj, v1.EventTypeWarning, string(alertmanagerv1alpha1.Error), "Unable to create/update status due to error "+err.Error())
```
Comment (lilwan) [inline controllers/common/common.go:72]: how does the event record handle different event type?

File: pkg/wavefront/api.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package wavefront
+
+import "context"
+
+type Wavefront struct {
+
+}
+
+//CreateOrUpdateWavefrontAlert creates/update a wavefront alert
+func (w *Wavefront)	CreateOrUpdateWavefrontAlert(ctx context.Context, input AlertInput) (output AlertOutput, err error) {
```
Comment (lilwan) [inline pkg/wavefront/api.go:10]: Assuming wavefront create/update alert APIs are the same?

File: controllers/common/common.go (hunk 1 above)
Comment (mnkg561) [reply controllers/common/common.go:72]: It doesn't have to handle anything different. K8s events are of 2 types - 1. Normal 2. Warning- When you do 
```k8s get events``` it needs that type so we are passing what type of event that is as part of event creation

```
mesh-health-check                     5m49s       Normal    SawCompletedJob           cronjob/mesh-health-check-cronjob                                (combined from similar events): Saw completed job: mesh-health-check-cronjob-1624644900, status: Complete
mesh-health-check                     11m         Warning   Unhealthy                 pod/mesh-health-check-deployment-f4b989b88-t7rjh                 Readiness probe failed: Get http://10.192.20.50:15021/healthz/ready: net/http: request canceled (Client.Timeout exceeded while awaiting headers)
```

File: pkg/wavefront/api.go (hunk 2 above)
Comment (mnkg561) [reply pkg/wavefront/api.go:10]: Yeah- That is the assumption for now

**Review by lilwan — APPROVED** [review]

**Pull request merged**